File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
# 그냥 1개 큰 값들로 정렬(회전시켜서)
# 그 다음 그 중 최댓값 X 반대값 중 최댓값 
    big = set()
    small = set()
    for length in sizes:
        big.add(max(length[0], length[1]))
        small.add(min(length[0], length[1]))
    return int(max(big)) * int(max(small))
        

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
import java.util.*;
class Solution {
    public int solution(int[][] sizes) {
        // 한쪽으로 큰 값들을 다 몰고
        // 다른 한쪽에 작은 값들 몰아
        int size = sizes.length;
        int[] row = new int[size];
        int[] col = new int[size];
        
        for(int i = 0; i < size; i++) {
            int[] temp = sizes[i];
            if(temp[0] < temp[1]) {
                row[i] = temp[1];
                col[i] = temp[0];
            } else {
                row[i] = temp[0];
                col[i] = temp[1];
            }
        }
        
        Arrays.sort(row);
        Arrays.sort(col);
        
        int answer = row[size-1]*col[size-1];
        return answer;
    }
}